### PR TITLE
add Cassandra sinks

### PR DESCRIPTION
**Pull request by nevillelyh**
Had to monkey patch a lot in `scio-cassandra2` due to an old Guava dependency. Otherwise 2 & 3 are sharing Scala code and handling API changes via 2 util classes.

Couldn't get `cassandra-unit` to work since `BulkCqlWriter` uses C* internal API and conflicts with embedded C* when running in the same JVM. Will probably need to fork it as integration test. Thoughts?

### CODE REVIEW

File: scio-cassandra3/src/main/java/com/spotify/scio/cassandra/CompatUtil.java [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+
+  private static RandomPartitioner randomPartitioner = new RandomPartitioner();
+  private static Murmur3Partitioner murmur3Partitioner = new Murmur3Partitioner();
+
+  public static ProtocolVersion getProtocolVersion(Cluster cluster) {
+    return cluster.getConfiguration().getProtocolOptions().getProtocolVersion();
+  }
+
+  public static <T> ByteBuffer serialize(DataType dataType, T value,
+                                     ProtocolVersion protocolVersion) {
```
Comment (andrewsmartin) [inline scio-cassandra3/src/main/java/com/spotify/scio/cassandra/CompatUtil.java]: Indentation

File: scio-cassandra3/src/main/java/org/apache/cassandra/hadoop/cql3/CqlBulkRecordWriterUtil.java [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    CqlBulkOutputFormat.setTableSchema(conf, table, tableSchema);
+    CqlBulkOutputFormat.setTableInsertStatement(conf, table, insertStatement);
+
+    // workaround for Hadoop static initialization
+    if (!System.getProperties().containsKey("hadoop.home.dir") &&
+        !System.getenv().containsKey("HADOOP_HOME")) {
+      System.setProperty("hadoop.home.dir", "/");
+    }
+
+    DatabaseDescriptor.clientInitialization();
```
Comment (andrewsmartin) [inline scio-cassandra3/src/main/java/org/apache/cassandra/hadoop/cql3/CqlBulkRecordWriterUtil.java]: What's this for? AFAICT this is the only difference from the same class in `scio-cassandra2`

Comment (nevillelyh) [inline scio-cassandra3/src/main/java/org/apache/cassandra/hadoop/cql3/CqlBulkRecordWriterUtil.java]: And
```scala
CqlBulkOutputFormat.setColumnFamilySchema(conf, table, tableSchema);
CqlBulkOutputFormat.setColumnFamilyInsertStatement(conf, table, insertStatement);
```

vs

```scala
CqlBulkOutputFormat.setTableSchema(conf, table, tableSchema);
CqlBulkOutputFormat.setTableInsertStatement(conf, table, insertStatement);
```

Comment (andrewsmartin) [inline scio-cassandra3/src/main/java/org/apache/cassandra/hadoop/cql3/CqlBulkRecordWriterUtil.java]: 👍 missed that

**Comment by nevillelyh** [comment]
Added integration test but only for `scio-cassandra3` since CircleCI supports Cassandra 3.4 only.

**Comment by nevillelyh** [comment]
Also this won't work very well in streaming mode since it does a `groupByKey` and only start writing at the end after `CqlBulkRecordWriter#close()`.

**Pull request merged**